src/Illuminate/Console/GeneratorCommand.php

```diff
@@ -83,6 +83,8 @@ protected function getPath($name)
 	 */
 	protected function parseName($name)
 	{
+		$name = strtok($name, '@');
```

Comment:
I totally disagree. `strtok` is exactly what we want here, and will perform better.